src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1793,21 +1833,23 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
                                  client.create_or_update,
                                  resource_group_name=resource_group_name,
                                  resource_name=name, parameters=mc)
+
             # add cluster spn with Monitoring Metrics Publisher role assignment to the cluster resource
             # mdm metrics supported only in azure public cloud so add the  role assignment only in this cloud
             cloud_name = cmd.cli_ctx.cloud.name
             if cloud_name.lower() == 'azurecloud' and monitoring:
+                # adding a wait here since we rely on the result for role assignment
+                result = LongRunningOperation(cmd.cli_ctx)(result)
```

Comment:
Per my understanding, waiting is necessary here. 

However, if `no_wait` is set to `True`, I don't think `LongRunningOperation` works anymore. In such case, `result` is a [`LROPoller`](https://github.com/Azure/msrest-for-python/blob/master/msrest/polling/poller.py#L103) whose `polling_method` is [`NoPolling`](https://github.com/Azure/msrest-for-python/blob/master/msrest/polling/poller.py#L66) whose [`finished()`](https://github.com/Azure/msrest-for-python/blob/master/msrest/polling/poller.py#L91) method always returns `True`. 

If you want to force `LongRunningOperation` to wait here, you need to remove `sdk_no_wait`. For example, during VM creation: 

https://github.com/Azure/azure-cli/blob/d3beab81c9245a3e815a08fe3034a89ba1b52f88/src/azure-cli/azure/cli/command_modules/vm/custom.py#L826-L830

Reply:
As discussed offline, made chnages as per our conversation